### ISSUE TITLE
Delete Folder while Uploading

### ISSUE THREAD

**Issue by Tloru** [keybase/client#19256]
Keybase GUI Version: 4.4.0-20190829183920+8f13068775
Using Keybase on Mac with 'Enable Keybase in Finder' enabled. 

Recently, I started uploading a 22+ GB folder to Keybase in one go... however, I realized that I had uploaded the wrong folder, and deleted it while it was uploading it's contents. This only partially worked, and left Keybase in a messy state ('still uploading' messages, hanging, errors, etc.) and didn't resolve without some fiddling and a restart (and even now it isn't fully fixed). This is probably due to the nature of some of the files I had in that folder, some over a few gigs in size. 

TL;DR: Cancelling large folder upload causes errors.

**Comment by strib** [comment]
Thanks for the report.  The file system has a goal of keeping an ordered record of all data written to it, so that you can recover what you've written, and also for low-level technical reasons involving the file system interface.  So when you upload a large amount of data, and then delete it, the upload of the original data still happens in the background, before the delete can take effect.  I definitely understand that there are scenarios like this where you've uploaded it by accident and want to cancel that -- we are still trying to figure out the best way to let you do that, within the restrictions of the file system interface.

If you're still having issues and want me to look at your specific situation, I'd be happy to.  You can share your logs with me using `keybase log send` from a terminal, and reporting the resulting log ID here.  I can take a look and give you some specific manual instructions for how to stabilize things.
